### DISCUSSION TITLE
IAM config provider for Jedis?

### DISCUSSION THREAD

**Discussion by ohookins** [xetorthio/jedis#3462] [Q&A]
We are considering using IAM authentication with AWS Elasticache / Redis; we're already using Jedis in at least one project. While it seems that Jedis supports injecting a config provider which could implement the username/password refreshing that would be required to operate with IAM-based credentials, I haven't been able to find an open source config provider for this. It's also not supported in the Jedis project itself.

Has anyone come across a project that implements this? I'm surprised I couldn't find any issues, PRs or discussions about this already but perhaps I'm searching for the wrong thing.

**Comment by yangbodong22011** [comment]
Hello, @ohookins you want a sample code for Jedis use IAM? or do you already have one?

**Reply by ohookins** [reply]
I guess so, and even better would be an integrated config provider for AWS in the codebase.
There are some examples in the AWS docs but they don't tend to do any renewal of credentials periodically.

**Comment by rustlingwind** [comment]
I also need an IAM credential provider for Jedis so that can use Jedis connect to AWS Elasticache which enabled the IAM authentication.
I haven't found any examples and not sure if Jedis has implemented this feature.
As our project has been using Jedis for loading data into Redis, so I am eager to get this example.
Oops, I can google many such examples of Lettuce, but cannot get one for Jedis.
